pkg/controller/perconaservermongodb/custom_users.go

```diff
@@ -150,6 +130,55 @@ func handleUsers(ctx context.Context, cr *api.PerconaServerMongoDB, mongoCli mon
 	return nil
 }
 
+func validateUser(ctx context.Context, user *api.User, sysUserNames, uniqueUserNames map[string]struct{}) error {
+	log := logf.FromContext(ctx)
+
+	if sysUserNames == nil || uniqueUserNames == nil {
+		log.Error(nil, "sys or unique usernames are nil")
```

Comment:
do we need these logs for errors? don't the error we return will be logged as reconciler error?

Reply:
The error logs that we see here are only added because they existed also in the existing implementation. More specifically, we used to do something like this 

````
		if _, ok := sysUserNames[user.Name]; ok {
			log.Error(nil, "creating user with reserved user name is forbidden", "user", user.Name)
			continue
		}
````

but now we do this:


```
		err := validateUser(ctx, &user, systemUserNames, uniqueUserNames)
		if err != nil {
			continue
		}
```

We can either keep the implementation as is, or we can add more details to the errors retuned and log a single time when we handle the `validateUser` error. Maybe the latter is more clean.

Reply:
Running this with errors, looks like this with the PR's implementation:

![Screenshot 2025-01-31 at 1 34 54 PM](https://github.com/user-attachments/assets/db94c848-5a12-45fa-9352-1f162c6cf889)